src/plugins_3rdparty/primer3/src/Primer3Task.h

```diff
@@ -29,45 +29,71 @@
 
 #include "FindExonRegionsTask.h"
 #include "Primer3TaskSettings.h"
-#include "primer3_core/primer3.h"
+#include "primer3_core/libprimer3.h"
 
-struct primers_t;
 
 namespace U2 {
 
 class AnnotationTableObject;
 
-class Primer {
+class PrimerSingle {
 public:
-    Primer();
-    Primer(const primer_rec& primerRec);
-    bool operator==(const Primer& primer) const;
-    static bool areEqual(const Primer* p1, const Primer* p2);
+    PrimerSingle(oligo_type type = oligo_type::OT_LEFT);
+    PrimerSingle(const primer_rec& primerRec, oligo_type type, int offset);
+    bool operator==(const PrimerSingle& primer) const;
+    static bool areEqual(const PrimerSingle* p1, const PrimerSingle* p2);
 
     int getStart() const;
     int getLength() const;
     double getMeltingTemperature() const;
+    double getBound() const;
     double getGcContent() const;
-    short getSelfAny() const;
-    short getSelfEnd() const;
+    double getSelfAny() const;
+    double getSelfEnd() const;
+    double getTemplateMispriming() const;
+    double getHairpin() const;
     double getEndStability() const;
+    double getQuality() const;
+    double getRepeatSim() const;
+    const QString& getRepeatSimName() const;
+    const QString& getSelfAnyStruct() const;
+    const QString& getSelfEndStruct() const;
+    oligo_type getType() const;
 
     void setStart(int start);
     void setLength(int length);
     void setMeltingTemperature(double meltingTemperature);
+    void setBound(double bound);
     void setGcContent(double gcContent);
-    void setSelfAny(short selfAny);
-    void setSelfEnd(short selfEnd);
+    void setSelfAny(double selfAny);
+    void setSelfEnd(double selfEnd);
+    void setTemplateMispriming(double templateMispriming);
+    void setHairpin(double hairpin);
     void setEndStability(double endStability);
+    void setQuality(double quality);
+    void setRepeatSim(double repeatSim);
+    void setRepeatSimName(const QString& repeatSimName);
+    void setSelfAnyStruct(const QString& selfAnyStruct);
+    void setSelfEndStruct(const QString& selfEndStruct);
 
 private:
+    // don't forget to change copy constructor and assignment operator when changing this!
```

Comment:
Why do you need copy constructor & assignment operator for trivial structs like this? The default ones should work + you can remove all the boilerplate with getters and setters and just make all fields public.

Reply:
Yep, this is some rudimental stuff from the previous version, I checked everything and upgraded:

`PrimerSingle` - has only primitive members, so do not need overloaded constructors/operators, default ones works well

`PrimerPair` - has pinters as members, should have overrided constructors/operators. Copy constructor and operator have been redefined, move constructor and operator have been deleted (they are not used, don't see a point in a real defenition), `operator==` has been deleted (it's not used, how I found out)